alibi_detect/base.py

```diff
@@ -102,16 +102,23 @@ class DriftConfigMixin:
     preprocess_fn: Optional[Callable] = None
 
     def drift_config(self):
+        name = self.__class__.__name__
+        # strip off any backend suffix
+        backends = ['TF', 'Torch', 'Sklearn']
+        for backend in backends:
+            if name.endswith(backend):
+                name = name[:-len(backend)]
```

Comment:
Doesn't feel right to have to do thi, but at the same time I have no other ideas :)

Reply:
Eurgh I know, I hate it. I wanted to do something clever with checking `self.__class__.__base__` or similar i.e. determine if `self` is a backend-specific detector by checking if its parent class is a detector itself (instead of a base class in `cd/base.py`. The problem is that `MMDDriftTF` doesn't actually inherit `MMDDrift`, since in `MMDDrift` we do `self._detector = MMDDrift()`. So when `MMDDriftTF.get_config()` is called by `MMDDrift.get_config()`, it thinks its parent class is `BaseMMDDrift`. Don't know if that makes sense?

Reply:
Can we also just overwrite it in the child class? So in the `get_config`we replace the `name` keys value with the correct class name which we know at that point in the code. It's still not a great way of solving it though. I think it's only better in the sense that the general behaviour is captured here and then deviations from that behaviour occur closer to the place they happen. If that makes sense?